### DISCUSSION TITLE
why encoding/v2 index file have multiply page?

### DISCUSSION THREAD

**Discussion by zdyj3170101136** [grafana/tempo#3058] [Q&A]
why encoding/v2, one index file would have multiply separate page, which provide checksum?

s3 already provide file consistence.

**Comment by joe-elliott** [comment]
v2 (and all Tempo encodings) are written to work on multiple types of object storage as well as local disk. The encoding does not make assumptions about the way it is stored.

**Comment by zdyj3170101136** [comment]
> v2（以及所有 Tempo 编码）被编写为可在多种类型的对象存储以及本地磁盘上工作。编码不会对其存储方式做出假设。

one index file contains one page is enough.

is the multiply page to provide checksum accuracy ?
or provide high decompress speed?(cause we could only decompress part of index file)

**Comment by zdyj3170101136** [comment]
@joe-elliott

**Comment by joe-elliott** [comment]
Paging allows a querier to have a consistent workload size regardless of the block size. It allows the querier to only load parts of the index instead of the entire thing.
